draft-trammell-taps-post-sockets.md

```diff
@@ -701,23 +719,83 @@ func receiveMulticast() {
 }
 ~~~~~~~~
 
+## Association Bootstrapping
+
+Here, we show how Association state may be initialized without a carrier.
+The goal is to create a long-term Association from which Carriers may be derived
+and, if possible, used immediately. Per {{I-D.ietf-taps-transport-security}},
+a first step is to specify trust model constraints, such as pinned public keys
+and anchor certificates, which are needed to create Remote connections.
+
+We begin by creating shared security parameters that will be used later for creating
+a remote connection.
+
+~~~~~~~~
+// create security parameters with a set of trusted certificates
+func createParameters(trustedCerts []Certificate) Parameters {
+    parameters := Parameters()
+    parameters = parameters.SetTrustedCerts(trustedCerts)
+    return parameters
+}
+~~~~~~~~
+
+Using these statically configured parameters, we now show how to create an Association
+between a Local and Remote using these parameters.
+
+~~~~~~~~
+// create an Association using shared parameters
+func createAssociation(local Local, remote Remote, parameters Parameters) Association {
+    association := AssociationWithParameters(local, remote, parameters)
+    return association
+}
+~~~~~~~~
+
```

Comment:
I'm wondering now if the certs should maybe be a property of the remote?

Reply:
Certs are probably properties of both Associations and Remotes. During  bootstrapping, a remote node may choose to present a different certificate for an association depending on peer identity (i.e., the Local/Remote pair). OTOH in the case of DANE, a certificate or certificate hash may be retrieved from the DNS as part of Remote resolution. I think we need both.

Reply:
These are the trusted certificate anchors that a Local would use to verify the certificates offered up by a remote node. And they're the same anchors used across all connections. You can certainly set specific a specific anchor (or leaf, to pin) on an Association or Remote, though that seemed like a detail to specify elsewhere.